stock_account_operating_unit/model/stock_move.py

```diff
@@ -0,0 +1,134 @@
+            ):
+                raise exceptions.UserError(
+                    _(
+                        "You cannot create stock moves involving separate source"
+                        " and destination accounts related to different "
+                        "operating units."
+                    )
+                )
+
+            if not self.operating_unit_dest_id and not self.operating_unit_id:
+                ou_id = (
+                    self.picking_id.picking_type_id.warehouse_id.operating_unit_id.id
+                )
+            else:
+                ou_id = False
+
+            debit_line_vals["operating_unit_id"] = (
+                ou_id or self.operating_unit_dest_id.id or self.operating_unit_id.id
+            )
+            credit_line_vals["operating_unit_id"] = (
+                ou_id or self.operating_unit_id.id or self.operating_unit_dest_id.id
+            )
+            rslt["credit_line_vals"] = credit_line_vals
+            rslt["debit_line_vals"] = debit_line_vals
+            if price_diff_line_vals:
+                price_diff_line_vals["operating_unit_id"] = (
+                    ou_id or self.operating_unit_id.id or self.operating_unit_dest_id.id
+                )
+                rslt["price_diff_line_vals"] = price_diff_line_vals
+        return rslt
+
+    def _action_done(self, cancel_backorder=False):
+        """
+        Generate accounting moves if the product being moved is subject
+        to real_time valuation tracking,
+        and the source or destination location are
+        a transit location or is outside of the company or the source or
+        destination locations belong to different operating units.
+        """
+        res = super(StockMove, self)._action_done(cancel_backorder)
+        for move in self:
+
+            if move.product_id.valuation == "real_time":
+                # Inter-operating unit moves do not accept to
+                # from/to non-internal location
+                if (
+                    move.location_id.company_id
+                    and move.location_id.company_id == move.location_dest_id.company_id
+                    and move.operating_unit_id != move.operating_unit_dest_id
+                ):
+                    (
+                        journal_id,
+                        acc_src,
+                        acc_dest,
+                        acc_valuation,
+                    ) = move._get_accounting_data_for_valuation()
+
+                    move_lines = move._prepare_account_move_line(
+                        move.product_qty,
+                        move.product_id.standard_price,
+                        acc_valuation,
+                        acc_valuation,
+                        _("%s - OU Move") % move.product_id.display_name,
+                    )
+                    am = (
+                        self.env["account.move"]
+                        .with_context(
```

Comment:
```suggestion
                        .sudo()
                        .with_context(
```



If you don't use this you must give billing access to stock users and some companies don't want this.

Odoo do it in this lines in stock_account module.

https://github.com/odoo/odoo/blob/14.0/addons/stock_account/models/stock_move.py#L455-L464